src/pgwire/src/protocol.rs

```diff
@@ -784,20 +779,17 @@ where
                     .await
             }
         };
-        self.conn
-            .send(BackendMessage::ParameterDescription(
-                stmt.desc()
-                    .param_types
-                    .iter()
-                    .map(mz_pgrepr::Type::from)
-                    .collect(),
-            ))
-            .await?;
+        // Cloning to avoid a mutable borrow issue because `send` also uses `coord_client`
+        let desc = stmt.desc().clone();
```

Comment:
I think probably better to hoist like so?

```rust
let message = BackendMessage::ParameterDescription(...);
self.send(message).await?;
```

Reply:
Ah, I managed to get it to work without cloning by building all the messages first and then later sending them at the end of the method, thanks for the suggestion!